src/store/modules/eth/actions.js

```diff
@@ -89,7 +90,7 @@ const createSpecificActions = (api) => ({
     void api.getGasPrice().then((price) => {
       // It is OK with London hardfork
       context.commit('gasPrice', {
-        gasPrice: price, // string type
+        gasPrice: Number(price), // string type
         fee: +(+utils.calculateFee(DEFAULT_ETH_TRANSFER_GAS, price)).toFixed(8) // number type, in ETH
```

Comment:
Why we use `DEFAULT_ETH_TRANSFER_GAS` here?

Reply:
We cannot use `api.estimateGas()` here because we don't have the tx data at this moment. This action is triggered in the background and the purpose of this is to estimate the tx fee and show it in the UI.

![image](https://github.com/Adamant-im/adamant-im/assets/25831507/807b7b6f-0816-443b-99c0-293049b553c3)

Reply:
Then create a tx in background and use `const gasLimit = api.estimateGas` or set to `CryptosInfo[${crypto}].defaultGasLimit` if unable to receive.
When we have no amount to send in the input field, calc as we send all the amount.
When we have no recipient in the input field, calc as we send to ourself.

Reply:
> When we have no amount to send in the input field, calc as we send all the amount.
When we have no recipient in the input field, calc as we send to ourself.

We can't do this without architectural change. Since we store input values (amount, recipient) in the component itself (SendFundsForm) we can't access this information from the store.

I suggest to skip this refactoring. Added the reference in https://trello.com/c/79uvtQU2/455-refactor-eth-erc20-store